Src/libCZI/SingleChannelAccessorBase.cpp

```diff
@@ -134,3 +134,39 @@ std::vector<int> CSingleChannelAccessorBase::CheckForVisibility(const libCZI::In
     std::reverse(result.begin(), result.end());
     return result;
 }
+
+/*static*/CSingleChannelAccessorBase::SubBlockData CSingleChannelAccessorBase::GetSubBlockDataForSubBlockIndex(
+    const std::shared_ptr<libCZI::ISubBlockRepository>& sbBlkRepository,
+    const std::shared_ptr<libCZI::ISubBlockCache>& cache,
+    int subBlockIndex)
+{
+    SubBlockData result;
+
+    // if no cache-object is given, then we simply read the subblock and create a bitmap from it
+    if (!cache)
```

Comment:
This be restructured return early on case of cache and cache hit and just get subblock normally otherwise:
```
if (cache)
  if (cache-hit)
    return result-with-bm-from-cache

get-subblock normally
return result-w-image-from-noncached-subblock
```

Reply:
...not sure what you are objecting in particular, but some words about my motivation:

* The method (in its current version) uses only a single return point, that's an effort in order to help the compiler apply the copy elision optimization (c.f. e.g. [here](https://stackoverflow.com/questions/12953127/what-are-copy-elision-and-return-value-optimization)). Granted, it's only a very minor performance benefit and probably a case of premature optimization, however it has become my habit.
* The operations for "no-cache-hit" and "no-cache-available" are actually not quite the same. A good question is whether they can and maybe should be made the same. Will have to think about it, but the initial version was "playing safe and leave existing behavior (i.e. the case "no cache") as it is".

Reply:
I added an additional parameter - leaving the user the choice whether uncompressed subblocks are also cached. It might make sense to not cache uncompressed subblocks (since we then may get with a fast storage/file-cache) similar performance than with the caching.